src/main/java/me/anxuiz/settings/bukkit/plugin/gui/clickables/Clickable.java

```diff
@@ -0,0 +1,28 @@
+package me.anxuiz.settings.bukkit.plugin.gui.clickables;
+
+import me.anxuiz.settings.bukkit.plugin.BukkitSettingsPlugin;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.ItemStack;
+
+public abstract class Clickable implements Click, Listener {
```

Comment:
I think you should reconsider the design choice of having each button register as a Bukkit listener. The Bukkit event system is not really designed to have lots of listeners for specific event targets. It assumes you will have a central listener and do your own dispatching.

Another problem with this system is that `ItemStack` is a value type, i.e. they compare equal if their contents are identical. So, you can't identify a specific window and slot just by comparing the contents, unless you encode a unique ID in the stack somehow.